src/lib/parseParams.ts

```diff
@@ -0,0 +1,200 @@
+import { cosmiconfig } from "cosmiconfig";
+import * as YAML from "yaml";
+import * as fs from "fs";
+
+type DenoifyParams = {
+    replacer?: string;
+    ports?: {
+        [portName: string]: string;
+    };
+    out?: string;
+    index?: string;
+    includes?: (
+        | string
+        | {
+              src: string;
+              destDir?: string;
+              destBasename?: string;
+          }
+    )[];
+};
+
+type ConfigFileType =
+    | {
+          type: "absent";
+      }
+    | {
+          type: "json" | "js" | "yaml" | "extensionless";
+          file: string;
+          config: string;
+      };
+
+function parseAsStringElseUndefined(param: unknown) {
+    return typeof param === "string" ? param : undefined;
+}
+
+function parseAsStringElseThrow({ param, type }: { param: unknown; type: string }) {
+    if (typeof param === "string") {
+        return param;
+    }
+    throw new Error(
+        [
+            "Denoify configuration Error",
+            `Expect ${type} to be string, got ${param} instead`,
+            "See: https://github.com/garronej/my_dummy_npm_and_deno_module"
+        ].join("\n")
+    );
+}
+
+function parseAsDenoifyParams(denoifyParams: any): DenoifyParams | undefined {
+    if (denoifyParams === undefined) {
+        return undefined;
+    }
+    const { includes } = denoifyParams;
+    return {
+        replacer: parseAsStringElseUndefined(denoifyParams.replacer),
+        out: parseAsStringElseUndefined(denoifyParams.out),
+        index: parseAsStringElseUndefined(denoifyParams.index),
+        includes: !Array.isArray(includes)
+            ? undefined
+            : includes.map(elem =>
+                  typeof elem === "string"
+                      ? elem
+                      : {
+                            destDir: parseAsStringElseUndefined(elem.destDir),
+                            destBasename: parseAsStringElseUndefined(elem.destBasename),
+                            src: parseAsStringElseThrow({
+                                param: elem.src,
+                                type: "src in includes array"
+                            })
+                        }
+              ),
+        ports:
+            denoifyParams.ports !== undefined || denoifyParams.ports !== null
+                ? undefined
+                : Object.entries(denoifyParams.ports).reduce(
+                      (prev, [portName, value]) => ({
+                          ...prev,
+                          [portName]: parseAsStringElseThrow({
+                              param: value,
+                              type: "value of ports object"
+                          })
+                      }),
+                      {}
+                  )
```

Comment:
```suggestion
        "replacer": parseAsStringElseUndefined(denoifyParams.replacer),
        "out": parseAsStringElseUndefined(denoifyParams.out),
        "index": parseAsStringElseUndefined(denoifyParams.index),
        "includes": !Array.isArray(includes)
            ? undefined
            : includes.map(elem =>
                  typeof elem === "string"
                      ? elem
                      : {
                            "destDir": parseAsStringElseUndefined(elem.destDir),
                            "destBasename": parseAsStringElseUndefined(elem.destBasename),
                            "src": parseAsStringElseThrow({
                                "param": elem.src,
                                "type": "src in includes array"
                            })
                        }
              ),
        "ports":
            denoifyParams.ports !== undefined || denoifyParams.ports !== null
                ? undefined
                : Object.entries(denoifyParams.ports).reduce(
                      (prev, [portName, value]) => ({
                          ...prev,
                          [portName]: parseAsStringElseThrow({
                              "param": value,
                              "type": "value of ports object"
                          })
                      }),
                      {}
                  )
```

Reply:
I think it helps for code readability to declare object with JSON syntax.  
This way it makes it clear what is a type declaration and what is an Object instanciation.